src/storm-pomdp/parser/PomdpSolveParser.cpp

```diff
@@ -0,0 +1,422 @@
+#include "PomdpSolveParser.h"
+#include "storm/adapters/RationalNumberAdapter.h"
+#include "storm/api/builder.h"
+#include "Utility.h"
+#include <fstream>
+#include <sstream>
+#include <string>
+#include <vector>
+#include <map>
+#include <unordered_set>
+#include <iostream>
+
+namespace storm {
+namespace pomdp {
+namespace parser {
+
+struct POMDP {
+    std::vector<std::string> states;
+    std::vector<std::string> actions;
+    std::vector<std::string> observations;
+    std::vector<double> start_probabilities;
+    std::map<std::string, std::map<std::string, double>> transitions;
```

Comment:
Try using ``std::unordered_map`` instead of ``std::map``, it might be more performant as we do not need an ordering